elephant/spike_train_generation.py

```diff
@@ -1060,5 +1063,77 @@ def compound_poisson_process(rate, A, t_stop, shift=None, t_start=0 * ms):
                 for cp in cpp]
             return cpp
 
+
 # Alias for the compound poisson process
 cpp = compound_poisson_process
+
+
+def homogeneous_poisson_process_with_refr_period(rate, refr_period=3.*ms,
+                                                 t_start=0.0 * ms,
+                                                 t_stop=1000.0 * ms,
+                                                 as_array=False):
+    """
+    Returns a spike train whose spikes are a realization of a Poisson process
+    with the given rate and refractory period starting at time `t_start` and
+    stopping time `t_stop`.
+
+    All numerical values should be given as Quantities, e.g. 100*Hz.
+
+    Parameters
+    ----------
+
+    rate : Quantity scalar with dimension 1/time
+           The rate of the discharge.
+    refr_period : Quantity scalar with dimension time
+                  The time period the after one spike no other spike is
+                  emitted.
+    t_start : Quantity scalar with dimension time
+              The beginning of the spike train.
+    t_stop : Quantity scalar with dimension time
+             The end of the spike train.
+    as_array : bool
+               If True, a NumPy array of sorted spikes is returned,
+               rather than a SpikeTrain object.
+
+    Raises
+    ------
+    ValueError : If one of `rate`, `refr_period`, `t_start` and `t_stop`
+                 is not of type `pq.Quantity`.
+
+    Examples
+    --------
+        >>> from quantities import Hz, ms
+        >>> spikes = homogeneous_poisson_process_with_refr_period(
+            50*Hz, 3*ms, 0*ms, 1000*ms)
+        >>> spikes = homogeneous_poisson_process_with_refr_period(
+            20*Hz, 5*ms, 5000*ms, 10000*ms, as_array=True)
+
+    """
+    if not isinstance(t_start, Quantity) or not isinstance(t_stop, Quantity):
+        raise ValueError("t_start and t_stop must be of type pq.Quantity")
+    if not isinstance(refr_period, Quantity):
+        raise ValueError("refr_period must be of type pq.Quantity")
+    if not isinstance(rate, Quantity):
+        raise ValueError("rate must be of type pq.Quantity")
+
+    if t_stop.units != t_start.units:
+        t_stop = t_stop.rescale(t_start.units)
+
+    rate = rate.rescale(1 / t_start.units).magnitude
+    duration = (t_stop-t_start).magnitude
+    mean_spike_count = rate * duration
+    spike_count = np.random.poisson(lam=mean_spike_count)
+
+    refr_period = refr_period.rescale(t_start.units).magnitude
+    eff_duration = duration - (spike_count - 1) * refr_period
+    if eff_duration <= 0.:
+        raise ValueError('The duration is too short to place this number of ' +
```

Comment:
I don't think we should raise the error here.

Reply:
The point is that there is an inherently limit of the firing rate, as 1/ refractory period. For example with a refractory period of 3 ms, a maximal firing rate of 333 Hz, would mean that there is just one spike every 3 ms. A higher rate can't be achieved with this refr. period. 

But this the number_of_spikes should follow a Poisson distribution, and can't just raise the error directly as a comparison of rate with the inverse of the refr. period. 

What I could do is only raise a warning and return a spiketrain with equally spaces spikes each separated by the refractory period.

What do you think?

Reply:
maybe it is better too state the error as such:
raise ValueError('The fire rate must be sufficiently smaller than the inverse of the refractory period, otherwise the spikes will merely follow each other equidistantly. ')

Reply:
@dizcza I agree with @pbouss actually. I think there is an upper limit for the rate. Maybe we do not understand what you mean?
@pbouss I would rewrite the error message though, the sentence is a bit complicated

Reply:
@stellalessandra I rewrote the error message. How do you think, does it fit now?

Reply:
It was hard for me to grasp the first Peter's version that's why I asked for help. Now I get it. If the current implementation is just a rewrite of the old one, then I admit I was wrong.
Thank you!